server/game/gameSystems/DiscardCardSystem.ts

```diff
@@ -0,0 +1,47 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, Location } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardProperties extends ICardTargetSystemProperties {}
+
+export class DiscardCardSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, IDiscardProperties> {
+    public override readonly name = 'discardCard';
+    public override readonly eventName = EventName.OnCardsDiscarded;
+
+    public override canAffect(card: Card | Card[], context: TContext, additionalProperties: Record<string, any> = {}): boolean {
+        /* TODO A bug currently leads to this sometimes being an array. Once the bug is fixed, we can expect this to
```

Comment:
Can you try removing this and see if it's okay now? The most recent PR I merged should fix it, if not then I need to do more investigation and you can proceed with this in

Reply:
Discussed offline. Still an issue

Reply:
nvm will be fixed